observer/observer.go

```diff
@@ -0,0 +1,96 @@
+package observer
+
+import (
+	"strconv"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+
+	_ "github.com/letsencrypt/boulder/observer/probers/dns"
+	_ "github.com/letsencrypt/boulder/observer/probers/http"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	statMonitors = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "obs_monitors",
+			Help: "details of each configured monitor",
+		},
+		[]string{"name", "kind", "valid"},
+	)
+	statObservations = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "obs_observations",
+			Help:    "details of each probe attempt",
+			Buckets: []float64{.001, .002, .005, .01, .05, .1, .5, 1, 2, 5, 10},
+		},
+		[]string{"name", "kind", "success"},
+	)
+)
+
+// Observer contains the parsed, normalized, and validated configuration
+// describing a collection of monitors and the metrics to be collected.
+type Observer struct {
+	Logger   blog.Logger
+	Metric   prometheus.Registerer
+	Monitors []*monitor
```

Comment:
I think in general, let's only export fields in structs when we actually need them. The public-facing API of an Observer is a) `.Start()`, and b) the metrics it exports. I don't think these need to be made available to anyone. Also notably, neither the `.Logger` nor the `.Metric` fields are ever accessed by anything, so they can probably just be removed.